app/services/assignments_service/scheduler.rb

```diff
@@ -19,27 +21,27 @@ def course_dates
     end
 
     def business_days_count
-      calendar.business_days_between(startdate, enddate)
+      calendar.business_days_between(startdate, enddate) + 1 # add 1 due to start/end date offsets
     end
 
     def course_days_count
-      enddate.to_date.mjd - startdate.to_date.mjd
+      enddate.to_date.mjd - startdate.to_date.mjd + 1 # add 1 to have inclusive last day of course
     end
 
     private
 
-    attr_reader :assignment_count, :startdate, :enddate, :days
-
+    attr_reader :assignment_count, :days
```

Comment:
What's the reason for having attribute readers defined underneath private?

Reply:
I can't speak to the reason that the `attr_readers` were originally defined as `private` by former engineers. 

We pulled a couple of them out and made them public `attr_accessors` for the sake of being able to test the functionality of the scheduler and it's role in distributing due dates sufficiently. 

My guess is that the original engineers wanted to have the ability to call functions to access the data but restrict the use of these reader methods to the `Scheduler` class. 

For example, rather than defining the private method
```
def assignment_count
  @assignment_count
end
```
they just made the `attr_reader :assignment_count` so that they could call `assignment_count` as they do in the `course_dates` method instead of calling `self.assignment_count` repeatedly and have it only accessible to the `Scheduler`